magic_the_gathering/phases/main.py

```diff
@@ -0,0 +1,30 @@
+from magic_the_gathering.game_state import GameState
+from magic_the_gathering.phases.base import Phase
+
+
+class MainPhase(Phase):
+    def __init__(self, name: str):
+        super().__init__(
+            name=name,
+            allows_sorcery_speed_when_stack_is_empty=True,
+        )
+
+    def run(self, game_state: GameState) -> GameState:
+        # TODO: Review the implementation of this method
```

Comment:
One idea is to have a generic method that we call a lot that's called smth like `player_having_priority_may_take_an_action` that we call a lot, probably in while loops, that takes the flag `allows_sorcery_speed_when_stack_is_empty`.
It's very generic but it might be necessary: players can take many actions at many points, the only difference in the main phase is that sorcery speed actions are allowed when the stack is empty.
We need to handle the stack, passing of priority, activating activated abilities etc at many points in the turn, like even between the draw phase and main phase 1